src/DeckGlMap/helpers/style.js

```diff
@@ -0,0 +1,243 @@
+	}
+}
+
+/**
+ * Return array of RGBA values, where A is from 0 to 255
+ * @param rgbColorArray {Array} Array of RGB values
+ * @param opacity {number} From 0 to 1
+ * @returns {Array}
+ */
+function getRgbaColorArray(rgbColorArray, opacity) {
+	if (opacity || opacity === 0) {
+		return [...rgbColorArray, Math.floor(opacity * 255)];
+	} else {
+		return rgbColorArray;
+	}
+}
+
+/**
+ * Get style for given feature
+ * @param style {Object} Style definition suitable for Deck
+ * @param feature {GeoJSONFeature}
+ * @returns {Object} style object
+ */
+function getStyleForFeature(style, feature) {
+	if (style.attributeStyles) {
+		if (style.attributeStyles.length === 1) {
+			return getStyleObjectForAttribute(
+				style.attributeStyles[0],
+				style.baseStyle,
+				feature.properties
+			);
+		} else {
+			// TODO merge styles if there are styles for more attributes
+		}
+	} else {
+		return style.baseStyle;
+	}
+}
+
+/**
+ * @param attributeStyleDefinition {Object} Style definition for given attribute key
+ * @param baseStyleDefinition {Object}
+ * @param attributes {Object} Feature attributes
+ * @return {Object} Style object for given attribute key
+ */
+function getStyleObjectForAttribute(
+	attributeStyleDefinition,
+	baseStyleDefinition,
+	attributes
+) {
+	if (attributes.hasOwnProperty(attributeStyleDefinition.attributeKey)) {
+		let value = attributes[attributeStyleDefinition.attributeKey];
+		if (value === null || value === undefined) {
+			return baseStyleDefinition;
+		} else {
+			if (attributeStyleDefinition.attributeClasses) {
+				return getStyleObjectForAttributeClasses(
+					attributeStyleDefinition.attributeClasses,
+					value
+				);
+			} else if (attributeStyleDefinition.attributeValues) {
+				return getStyleObjectForAttributeValues(
+					attributeStyleDefinition.attributeValues,
+					value
+				);
+			}
+			// TODO add other cases
+			else {
+				return baseStyleDefinition;
+			}
+		}
+	} else {
+		return baseStyleDefinition;
+	}
+}
+
+// TODO export below methods from ptr-utils
+/**
+ * Attribute classes
+ *
+ * @param attributeClasses {Array}
+ * @param value {number|String} attribute value
+ * @return {Object} style object
+ */
+function getStyleObjectForAttributeClasses(attributeClasses, value) {
+	let styleObject = {};
+	_forEach(attributeClasses, attributeClass => {
+		let {interval, intervalBounds} = attributeClass;
+
+		if (!intervalBounds) {
+			intervalBounds = [true, false];
+		}
+
+		if (
+			isGreaterThan(value, interval[0], intervalBounds[0]) &&
+			isGreaterThan(interval[1], value, intervalBounds[1])
+		) {
+			styleObject = attributeClass;
+		}
+	});
+
+	return styleObject;
+}
+
+/**
+ * Attribute value
+ *
+ * @param attributeValues {Object}
+ * @param value {String} attribute value
+ * @return {Object}
+ */
+function getStyleObjectForAttributeValues(attributeValues, value) {
+	return attributeValues[value] || {};
+}
+
+function isGreaterThan(comparedValue, referenceValue, allowEquality) {
```

Comment:
Would probably be more readable to use function for determining if value is in interval, instead of two isGreaterThan